pkg/kosmosctl/floater/check.go

```diff
@@ -199,60 +203,90 @@ func (o *CommandCheckOptions) Run() error {
 func (o *CommandCheckOptions) RunRange(iPodInfos []*FloatInfo, jPodInfos []*FloatInfo) []*PrintCheckData {
 	var resultData []*PrintCheckData
 
+	goroutinePool := make(chan struct{}, o.routinesMaxNum)
+	waitGroup := sync.WaitGroup{}
+
 	if len(iPodInfos) > 0 && len(jPodInfos) > 0 {
 		for _, iPodInfo := range iPodInfos {
 			for _, jPodInfo := range jPodInfos {
 				for _, ip := range jPodInfo.PodIPs {
-					var targetIP string
-					var err error
-					var cmdResult *command.Result
-					if o.DstFloater != nil {
-						targetIP, err = netmap.NetMap(ip, o.DstFloater.CIDRsMap)
-					} else {
-						targetIP = ip
-					}
-					if err != nil {
-						cmdResult = command.ParseError(err)
-					} else {
-						// ToDo RunRange && RunNative func support multiple commands, and the code needs to be optimized
-						cmdObj := &command.Ping{
-							TargetIP: targetIP,
+					goroutinePool <- struct{}{}
+					waitGroup.Add(1)
+					routineIPodInfo := iPodInfo
+					routineJPodInfo := jPodInfo
+					routineIp := ip
+					go func() {
```

Comment:
We  can definite a func to extract the logic. like go XXXX()

Reply:
i fixed it